app/scraper/sources/face_book.py

```diff
@@ -23,15 +23,43 @@
     MAJOR_FULL_NAMES = json.load(f)
 
 
+# Define a function to check if S3 credentials are set
+def has_s3_credentials():
+    return os.environ.get("S3_ACCESS_KEY") and os.environ.get("S3_SECRET_ACCESS_KEY")
```

Comment:
Put this in source.py in the base class that all the sources inherit from

Reply:
Actually put it in ImageUploader

Reply:
That won't work because neither source.py or ImageUploader are inherited by both of the places where has_s3_credentials() is used. I moved the one in face_book.py to source.py so it can be used in any future sources files

Reply:
and left the other one in the original class as a local utility function

Reply:
could also be replaced with two if() statements in those locations if simpler